schematic/exceptions.py

```diff
@@ -53,3 +53,28 @@ def __init__(self, arg_name: str,
 
     def __str__(self):
         return f"{self.message}"
+
+
+class AccessCredentialsError(Exception):
+    """Exception raised when provided access credentials cannot be resolved.
+    
+    Args:
+        project: Platform/project (e.g., synID of a project)
+        message: custom/pre-defined error message to be returned.
+
+    Returns:
+        message.
+    """
+
+    def __init__(self, project: str, message: str = None) -> str:
+        if message is not None:
+            self.message = message
+        else:
```

Comment:
You can avoid this else statement by setting the message to the default value before the if statement and then overwrite the message with user-provided value `if message:`.

Reply:
Great. This is cleaner, yes. I've incorporated this change.